### PR TITLE
test: address intermittent test failures

### PR DESCRIPTION
**Pull request by ykim-1**
## 📝 Description

Addressing some failures on TOD due to timeouts and behavior changes

Failures in TOD: http://198.19.5.79:1212/builds/6614a12400a8200001706548?team=DX&buildName=linode_api4&bld_id=6614a12400a8200001706548&testcaseStatus=failure

## ✔️ How to Test

`make testint` - should be all passing locally

## 📷 Preview

**If applicable, include a screenshot or code snippet of this change. Otherwise, please remove this section.**

### CODE REVIEW

**Review by yec-akamai — COMMENTED** [review]
I'm getting some different errors on my local, maybe just related to my account limitation:
```
=========================== short test summary info ============================
FAILED test/integration/models/test_linode.py::test_linode_create_disk - linode_api4.errors.ApiError: 400: [size] You do not have enough unallocated...
FAILED test/integration/models/test_linode.py::TestNetworkInterface::test_delete_interface_containing_vpc - TimeoutError: Timeout Error: resource is not available in 60 seconds
============ 2 failed, 117 passed, 21 skipped in 2462.97s (0:41:02) ============
```

File: test/integration/models/test_linode.py [hunk 1]
```diff
@@ -138,6 +138,10 @@ def get_status(linode: Instance, status: str):
     return linode.status == status
 
 
+def get_type(linode: Instance, type: str):
```
Comment (lgarber-akamai) [inline test/integration/models/test_linode.py:141]: Just to make things a bit clearer, could this be renamed to something like `instance_type_condition`?

File: test/integration/models/test_linode.py [hunk 2]
```diff
@@ -311,18 +316,38 @@ def test_linode_resize_with_migration_type(
...
+        wait_for_condition(
+            10,
+            60,
+            get_type,
+            test_linode_client.load(Instance, linode.id),
+            "g6-standard-1",
+        )
 
-    wait_for_condition(10, 300, get_status, linode, "resizing")
+        time.sleep(100)
```
Comment (lgarber-akamai) [inline test/integration/models/test_linode.py:334]: Just curious, what's the purpose of this sleep statement?

**Comment by lgarber-akamai** [comment]
> I'm getting some different errors on my local, maybe just related to my account limitation:
> 
> ```
> =========================== short test summary info ============================
> FAILED test/integration/models/test_linode.py::test_linode_create_disk - linode_api4.errors.ApiError: 400: [size] You do not have enough unallocated...
> FAILED test/integration/models/test_linode.py::TestNetworkInterface::test_delete_interface_containing_vpc - TimeoutError: Timeout Error: resource is not available in 60 seconds
> ============ 2 failed, 117 passed, 21 skipped in 2462.97s (0:41:02) ============
> ```

I'm seeing the `400: [size] ...` error on my end too, I'm thinking the initial disk from the fixture isn't being deleted now that the fixture isn't session-scoped

File: test/integration/models/test_linode.py (hunk 2 above)
Comment (ykim-1) [reply test/integration/models/test_linode.py:334]: So there is no "resizing" status anymore hence I manually ran the test steps couple times and found out that it takes about ~1 min to complete resizing

File: test/integration/models/test_database.py [hunk 3]
```diff
@@ -101,9 +101,9 @@ def test_get_types(test_linode_client):
     client = test_linode_client
     types = client.database.types()
 
-    assert (types[0].type_class, "nanode")
```
[the remark this answers is not in the document]
Comment (ykim-1) [reply test/integration/models/test_database.py:104]: Oh it was giving assertion warning so I changed up the syntax

File: test/integration/models/test_linode.py (hunk 2 above)
Comment (lgarber-akamai) [reply test/integration/models/test_linode.py:334]: Gotcha, thanks for clarifying! Would it make sense to [poll](https://github.com/linode/linode_api4-python/blob/dev/linode_api4/polling.py) for the `linode_resize` event instead?

**Comment by ykim-1** [comment]
https://github.com/linode/linode_api4-python/actions/workflows/e2e-test-pr.yml

**Review by lgarber-akamai — APPROVED** [review]
All tests are passing on my end, excellent work!

**Review by yec-akamai — APPROVED** [review]
Great work! All tests are passing now 🎉

**Pull request merged**